activation/validation_test.go

```diff
@@ -621,3 +616,27 @@ func validateNIPost(minerID types.NodeID, commitmentAtx types.ATXID, nipost *typ
 	_, err := v.NIPost(minerID, commitmentAtx, nipost, challenge, numUnits, opts...)
 	return err
 }
+
+func TestValidateMerkleProof(t *testing.T) {
+	proof := newMerkleProof(t, types.BytesToHash([]byte("challenge")), []types.Hash32{
+		types.BytesToHash([]byte("leaf2")),
+		types.BytesToHash([]byte("leaf3")),
+		types.BytesToHash([]byte("leaf4")),
+	})
+
+	t.Run("valid proof", func(t *testing.T) {
+		t.Parallel()
+
+		err := validateMerkleProof(proof)
+		require.NoError(t, err)
+	})
+	t.Run("invalid proof", func(t *testing.T) {
+		t.Parallel()
+
+		invalidProof := *proof
```

Comment:
I'm not sure but isn't this possibly racy?

Reply:
No, as long as `proof` is readonly.